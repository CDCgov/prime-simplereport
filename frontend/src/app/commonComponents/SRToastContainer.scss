.Toastify {
  &__toast-container {
    max-width: 40rem;
    width: auto;
  }

  &__toast {
    padding: 0;
  }

  &__close-button {
    height: 1.5rem;
    margin-top: 0.5rem;
    margin-right: 0.5rem;
    opacity: 1;
    width: 1.5rem;

    &:hover {
      opacity: 0.3;
    }

    svg {
      width: 100%;
      height: 100%;
    }
  }

  .sr-alert {
    &--error {
      background-color: #f4e3db;
    }

    &--success {
      background-color: #ecf3ec;

      .usa-alert--success {
        outline: none;
      }
    }
  }

  .usa-alert {
    max-width: 37.5rem;
  }
}
