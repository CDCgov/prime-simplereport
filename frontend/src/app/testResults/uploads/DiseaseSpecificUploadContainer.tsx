import React from "react";

import { FileUploadService } from "../../../fileUploadService/FileUploadService";
import { getAppInsights } from "../../TelemetryService";

import UploadForm from "./UploadForm";

const DiseaseSpecificUploadContainer = () => {
  const appInsights = getAppInsights();

  const alert = (
    <div className="usa-alert usa-alert--info margin-left-105em margin-right-105em maxw-tablet-lg">
      <div className="usa-alert__body">
        <h2 className="usa-alert__heading">
          New: Report flu and RSV results to California
        </h2>
        <div className="usa-alert__text">
          <p>
            Organizations sending results to the California Department of Public
            Health (CDPH) can now use the bulk results upload feature to report
            positive flu and RSV tests. Report COVID-19, flu A and B, and RSV
            results all from a single spreadsheet.{" "}
            <a
              href="https://www.simplereport.gov/assets/resources/bulk_results_upload_guide-flu_pilot.pdf"
              onClick={() => {
                appInsights?.trackEvent({
                  name: "Access bulk uploader flu guide",
                });
              }}
            >
              See more information and guidance about flu and RSV reporting
            </a>
            .
          </p>
          <p>
            All organizations can continue to report COVID-19 results to their
            public health department using the bulk uploader. Flu and RSV
            reporting will be available in other jurisdictions soon.
          </p>
        </div>
      </div>
    </div>
  );

  return (
    <UploadForm
      alert={alert}
      uploadResults={FileUploadService.uploadResults}
      uploadType={"Disease Specific"}
      spreadsheetTemplateLocation={
        "/assets/resources/test_results_example_10-3-2022.csv"
      }
      uploadGuideLocation={"/results/upload/submit/guide"}
    />
  );
};
export default DiseaseSpecificUploadContainer;
