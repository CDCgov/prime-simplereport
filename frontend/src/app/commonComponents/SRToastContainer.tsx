import React from "react";
import { ToastContainer } from "react-toastify";

import "react-toastify/dist/ReactToastify.css";
import "./SRToastContainer.scss";

const SRToastContainer = () => {
  const autoClose = 5000;
  const closeButton = true;
  const hideProgressBar = true;
  const limit = 2;
  const position = "bottom-left";

  return (
    <ToastContainer
      autoClose={autoClose}
      closeButton={closeButton}
      hideProgressBar={hideProgressBar}
      limit={limit}
      position={position}
    />
  );
};

export default SRToastContainer;
