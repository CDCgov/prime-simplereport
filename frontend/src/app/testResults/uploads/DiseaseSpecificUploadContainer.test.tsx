import { configureAxe } from "jest-axe";
import { render } from "@testing-library/react";
import createMockStore from "redux-mock-store";
import { Provider } from "react-redux";
import { MemoryRouter } from "react-router-dom";
import React from "react";

import SRToastContainer from "../../commonComponents/SRToastContainer";

import DiseaseSpecificUploadContainer from "./DiseaseSpecificUploadContainer";

const mockStore = createMockStore([]);
const store = mockStore();
const axe = configureAxe({
  rules: {
    // disable landmark rules when testing isolated components.
    region: { enabled: false },
  },
});

describe("Disease specific upload container test", () => {
  it("passes axe tests", async () => {
    render(
      <Provider store={store}>
        <MemoryRouter>
          <SRToastContainer />
          <DiseaseSpecificUploadContainer />
        </MemoryRouter>
      </Provider>
    );
    expect(await axe(document.body)).toHaveNoViolations();
    expect(document.body).toMatchSnapshot();
  });
});
