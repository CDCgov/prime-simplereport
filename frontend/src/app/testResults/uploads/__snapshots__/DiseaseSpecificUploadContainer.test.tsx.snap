// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Disease specific upload container test passes axe tests 1`] = `
<body>
  <div>
    <div
      class="Toastify"
    />
    <div
      class="grid-row header-size-fix sr-test-results-uploads"
    >
      <div
        class="prime-container card-container"
      >
        <div
          class="usa-card__header"
        >
          <h1>
            Upload your results
          </h1>
        </div>
        <div
          class="usa-alert usa-alert--info margin-left-105em margin-right-105em maxw-tablet-lg"
        >
          <div
            class="usa-alert__body"
          >
            <h2
              class="usa-alert__heading"
            >
              New: Report flu and RSV results in select jurisdictions
            </h2>
            <div
              class="usa-alert__text"
            >
              <p>
                SimpleReport allows organizations to report
                 
                COVID-19, Flu A and B, and RSV
                 results all from a single spreadsheet to a growing number of jurisdictions. To check if your jurisdiction is currently supported, please refer to the "Test for Other Diseases" section in the
                 
                <a
                  href="https://www.simplereport.gov/using-simplereport/test-for-other-diseases/influenza/"
                >
                  SimpleReport online user guide
                </a>
                .
                 
                <a
                  href="https://www.simplereport.gov/assets/resources/bulk_results_upload_guide-flu_pilot.pdf"
                >
                  See more information and guidance about reporting these diseases
                </a>
                .
              </p>
              <p>
                All organizations can continue to report COVID-19 results to their public health department using the bulk uploader. Flu and RSV reporting will be available in other jurisdictions soon.
              </p>
            </div>
          </div>
        </div>
        <div
          class="usa-card__body padding-y-2 maxw-prose"
        >
          <p>
            Report results in bulk using a comma-separated values (CSV) spreadsheet. To upload your spreadsheet:
          </p>
          <section>
            <ol
              class="usa-process-list"
            >
              <li
                class="usa-process-list__item margin-bottom-1em"
              >
                <p
                  class="usa-process-list__heading"
                >
                  Visit the
                   
                  <a
                    class=""
                    href="/results/upload/submit/guide"
                  >
                    <strong>
                      spreadsheet upload guide
                    </strong>
                  </a>
                </p>
              </li>
              <li
                class="usa-process-list__item margin-bottom-1em"
              >
                <p
                  class="usa-process-list__heading"
                >
                  Download the
                   
                  <a
                    href="/assets/resources/test_results_example_10-3-2022.csv"
                  >
                    spreadsheet template
                  </a>
                </p>
              </li>
              <li
                class="usa-process-list__item margin-bottom-1em"
              >
                <p
                  class="usa-process-list__heading"
                >
                  Following the guide and template, format your data to match SimpleReport requirements
                </p>
              </li>
              <li
                class="usa-process-list__item margin-bottom-1em"
              >
                <p
                  class="usa-process-list__heading"
                >
                  Save your spreadsheet in a CSV format (file size limit is 50 MB or 10,000 rows)
                </p>
              </li>
              <li
                class="usa-process-list__item margin-bottom-1em"
              >
                <p
                  class="usa-process-list__heading"
                >
                  Submit your CSV to the uploader below
                </p>
              </li>
            </ol>
          </section>
          <div
            class="usa-form-group margin-bottom-3"
            data-testid="formGroup"
          >
            <div
              class="sr-single-file-input blank"
            >
              <input
                accept="text/csv, .csv"
                aria-describedby="file-input-specific-hint"
                aria-invalid="false"
                aria-label="Choose CSV file"
                class="usa-file-input"
                data-testid="upload-csv-input"
                id="upload-csv-input"
                name="upload-csv-input"
                required=""
                type="file"
              />
              <span
                id="file-input-specific-hint"
              >
                Drag file here or choose from folder
              </span>
            </div>
          </div>
          <button
            class="usa-button"
            data-testid="button"
            disabled=""
            type="submit"
          >
            <span>
              Upload your CSV
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</body>
`;
